crates/environ/src/module.rs

```diff
@@ -311,6 +272,149 @@ impl Default for MemoryInitialization {
     }
 }
 
+#[allow(missing_docs)]
+pub enum InitMemory {
+    Ok,
+    NoGlobal,
+    NoImportedMemory,
+    OutOfBounds,
+    Error,
+}
+
+impl InitMemory {
+    /// Returns whether this is `InitMemory::Ok`
+    pub fn is_ok(&self) -> bool {
+        match self {
+            InitMemory::Ok => true,
+            _ => false,
+        }
+    }
+}
+
+impl MemoryInitialization {
+    /// Performs the memory initialization steps for this set of initializers.
+    ///
+    /// This will perform wasm initialization in compliance with the wasm spec
+    /// and how data segments are processed. This doesn't need to necessarily
+    /// only be called as part of initialization, however, as it's structured to
+    /// allow learning about memory ahead-of-time at compile time possibly.
+    ///
+    /// The various callbacks provided here are used to drive the smaller bits
+    /// of initialization, such as:
+    ///
+    /// * `get_cur_page_size` - gets the current size, in wasm pages, of the
```

Comment:
`get_cur_size_pages`  or `get_cur_size_in_pages` maybe? `get_cur_page_size` is otherwise ambiguous to me ("sure, the current page size is 64KiB").